### PR TITLE
Apply P0579R1: Constexpr for <experimental/ranges/iterator>

### LINKED ISSUE #256: Add constexpr to advance, distance, next, and prev
Also common_iterator(?).

### PR DESCRIPTION
**Pull request by CaseyCarter**
Fixes #256.

### CODE REVIEW

File: utilities.tex [hunk 1]
```diff
@@ -785,28 +792,28 @@
     tagged(const tagged&) = default;
     tagged &operator=(tagged&&) = default;
     tagged &operator=(const tagged&) = default;
-    tagged(Base&&) noexcept(@\seebelow@)
+    constexpr tagged(Base&&) noexcept(@\seebelow@)
       requires MoveConstructible<Base>;
-    tagged(const Base&) noexcept(@\seebelow@)
+    constexpr tagged(const Base&) noexcept(@\seebelow@)
```
Comment (ericniebler) [inline utilities.tex]: You thought you could sneak these by me, didn't you? ;-) P0579 doesn't specify these two to be `constexpr`. These constructors came from the resolution of #172, which also doesn't specify them to be `constexpr`. If we're playing by the book, we should leave the `constexpr` off here and file an issue. I'm content to look the other way if you wanted to leave them in, though.

File: iterators.tex [hunk 2]
```diff
@@ -244,48 +244,47 @@
...
       const reverse_iterator<I1>& x,
       const reverse_iterator<I2>& y);
   template <RandomAccessIterator I>
-    reverse_iterator<I>
-      operator+(
-    difference_type_t<I> n,
-    const reverse_iterator<I>& x);
+    constexpr reverse_iterator<I> operator+(
+      difference_type_t<I> n,
+      const reverse_iterator<I>& x);
```
Comment (ericniebler) [inline iterators.tex]: Wow that was some phenomenally bad formatting right there.

File: iterators.tex [hunk 3]
```diff
@@ -3723,38 +3717,38 @@
...
     constexpr move_sentinel();
-    explicit move_sentinel(S s);
-    move_sentinel(const move_sentinel<ConvertibleTo<S>>& s);
-    move_sentinel& operator=(const move_sentinel<ConvertibleTo<S>>& s);
+    explicit constexpr move_sentinel(S s);
+    constexpr move_sentinel(const move_sentinel<ConvertibleTo<S>>& s);
+    constexpr move_sentinel& operator=(const move_sentinel<ConvertibleTo<S>>& s);
 
-    S base() const;
+    constexpr S base() const;
```
Comment (ericniebler) [inline iterators.tex]: I also don't see these changes to `move_sentinel` in p0579. It's a screwup for sure, but we really should file a defect for this.

File: iterators.tex (hunk 2 above)
Comment (CaseyCarter) [inline iterators.tex]: Yeah, no idea how that happened.

File: utilities.tex (hunk 1 above)
Comment (CaseyCarter) [inline utilities.tex]: I was hoping to get away with these two; it's clearly in line with the intent of the change.

File: iterators.tex (hunk 3 above)
Comment (CaseyCarter) [inline iterators.tex]: Yes, these are too big a stretch. I'll pull these out and file an issue.

**Comment by CaseyCarter** [comment]
Repushed without the unauthorized changes to `move_sentinel`'s member functions.

**Comment by CaseyCarter** [comment]
...and without the changes to `tagged`'s constructors that accept `Base`.

**Pull request merged**